syntax = "proto3";

import "streamlit/proto/Arrow.proto";
import "streamlit/proto/ArrowNamedDataSet.proto";

message ArrowVegaLiteChart {
  // The a JSON-formatted string with the Vega-Lite spec.
  string spec = 1;

  // The dataframe that will be used as the chart's main data source, if
  // specified using Vega-Lite's inline API.
  Arrow data = 2;

  // Dataframes associated with this chart using Vega-Lite's datasets API, if
  // any.
  repeated ArrowNamedDataSet datasets = 4;

  // If True, will overwrite the chart width spec to fit to container.
  bool use_container_width = 5;

  reserved 3;
}
